scripts/content-types/load-content-types.ts

```diff
@@ -0,0 +1,252 @@
+/* eslint @typescript-eslint/no-var-requires: "off" */
+const fs = require('fs');
+const readline = require('readline');
+
+const JWP_HOST = process.env.JWP_HOST || 'https://api.jwplayer.com';
+
+interface SchemaFile {
+  fields: {
+    [key: string]: Field;
+  };
+  sections: {
+    [key: string]: Section<Field | string>;
+  };
+  schemas: Schema<Field | string, Section<Field | string> | string>[];
+}
+
+// TODO: Use yup to validate schema
+interface Field {
+  label: string;
+  param: string;
+  required?: boolean;
+  details: {
+    field_type: 'input' | 'toggle';
+    placeholder: string;
+  };
+}
+
+interface Section<TField extends Field | string> {
+  title: string;
+  fields: TField[];
+}
+
+interface Schema<TField extends Field | string, TSection extends Section<TField> | string> {
+  name: string;
+  description: string;
+  display_name: string;
+  is_series?: boolean;
+  is_active?: boolean;
+  hosting_type: 'hosted' | 'external' | 'ott_data';
+  sections: TSection[];
+}
+
+function error(message: string, returnCode?: number): never {
+  console.error(message);
+  process.exit(returnCode === undefined ? -1 : returnCode);
+}
+
+async function confirmPrompt(message: string, confirmMessage: string = 'Do you want to continue?') {
+  const stdIn = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  try {
+    stdIn.setPrompt(`${message}\r\n${confirmMessage} (y/N) `);
+    stdIn.prompt();
+    const response = await stdIn[Symbol.asyncIterator]().next();
+
+    if (response.done) {
+      process.exit(1);
+    }
+
+    return response.value?.toLowerCase() === 'y';
+  } finally {
+    stdIn.close();
+  }
+}
+
+function help() {
+  console.info('\r\nThis script can be used to upload content types to your JWP environment');
+  console.info('\r\n\r\nUsage: ts-node load-content-types --site-id=<JWP site ID> --api-key=<JWP V2 API Key> [optional arguments]');
+  console.info('\r\n\r\nRequired argument: site-id - specify the JWP property ID to upload to');
+  console.info('Required argument: api-key - specify the V2 API key found on the API Credentials page of the JWP dashboard');
+  console.info('Optional argument: source-file - specify a schema file to upload (defaults to ./content-types.json');
+}
+
+function parseSchemas(file: string) {
+  try {
+    const schemaDefinitions = JSON.parse(fs.readFileSync(file, 'utf-8')) as SchemaFile;
+
+    if (schemaDefinitions.schemas.length <= 0) {
+      error(`Error! No schemas found in file ${file}`, -4);
+    }
+
+    const mapSchema = (schema: Schema<Field | string, Section<Field | string> | string>) => {
```

Comment:
Can be probably used as a separate type (the same is used in SchemaFile['schemas'])

Reply:
Nice suggestion. Done.